llvm/lib/CodeGen/AsmPrinter/AsmPrinter.cpp

```diff
@@ -1982,9 +1982,10 @@ void AsmPrinter::emitFunctionBody() {
         break;
       default:
 
+        // Note that a patchpoint call, unlike a stackmap call, does actually
```

Comment:
I *think* I understand this comment in the context of this PR but after this PR is merged, this comment will probably bemuse readers, who won't have the context that makes it make sense.

I wonder if it should be reworded to something vaguely along the lines of "Stackmaps and statepoints have to XYZ. However, patchpoints -- even though they're similar to stackmaps and statepoints -- don't have to XYZ because ABC."?

Reply:
Maybe: "We don't need emit labels for STACKMAP since this intrinsic doesn't codegen a call. Patchpoints however, do emit calls, and so need to be included".

By the way, STATEPOINT also emits calls. But we don't use them.

Reply:
Does that mean that the `STATEPOINT` line in the conditional should also be removed?

Reply:
I think we can remove it. Since we don't use STATEPOINT it doesn't affect us, but technically I think it being there is incorrect. At least from what I can tell from https://llvm.org/docs/Statepoints.html#safepoint-semantics-verification.

Reply:
Let's just rip out the `STATEPOINT` line now; if we need it, we'll surely see it in some tests.

Reply:
Done in ea8d55c90a7304f04c1c313d80173b7dd282d267